builder/dockerfile/builder.go

```diff
@@ -286,6 +293,17 @@ func (b *Builder) build(stdout io.Writer, stderr io.Writer, out io.Writer) (stri
 		return "", fmt.Errorf("No image was generated. Is your Dockerfile empty?")
 	}
 
+	if b.options.Squash {
+		var fromID string
+		if b.from != nil {
```

Comment:
This would happen only when the parent is `scratch`, and in such cases it would squash to scratch, so there would be only a single layer in the image.